code/game/objects/structures/cyberplants.dm

```diff
@@ -16,44 +17,136 @@
 		"plant-13",
 		"plant-xmas",
 	)
+	var/global/list/possible_colors = list(
+		rgb(60, 148, 197),
+		COLOR_NAVY_BLUE,
+		COLOR_GREEN,
+		COLOR_DARK_GRAY,
+		COLOR_MAROON,
+		COLOR_PURPLE,
+		COLOR_VIOLET,
+		COLOR_OLIVE,
+		COLOR_BROWN_ORANGE,
+		COLOR_DARK_ORANGE,
+		COLOR_SEDONA,
+		COLOR_DARK_BROWN,
+		COLOR_BLUE,
+		COLOR_DEEP_SKY_BLUE,
+		COLOR_LIME,
+		COLOR_CYAN,
+		COLOR_TEAL,
+		COLOR_RED,
+		COLOR_PINK,
+		COLOR_ORANGE,
+		COLOR_YELLOW,
+		COLOR_BROWN,
+		COLOR_SUN,
+		COLOR_BLUE_LIGHT,
+		COLOR_RED_LIGHT,
+		COLOR_BEIGE,
+		COLOR_LUMINOL,
+		COLOR_SILVER,
+		COLOR_WHITE,
+		COLOR_NT_RED,
+		COLOR_BOTTLE_GREEN,
+		COLOR_GUNMETAL,
+		COLOR_CHESTNUT,
+		COLOR_BEASTY_BROWN,
+		COLOR_WHEAT,
+		COLOR_CYAN_BLUE,
+		COLOR_LIGHT_CYAN,
+		COLOR_PAKISTAN_GREEN,
+		COLOR_HULL,
+		COLOR_AMBER,
+		COLOR_COMMAND_BLUE,
+		COLOR_SKY_BLUE,
+		COLOR_CIVIE_GREEN,
+		COLOR_TITANIUM,
+		COLOR_DARK_GUNMETAL,
+	)
 
 /obj/structure/cyberplant/New()
 	..()
-	plant = prepare_icon(plant)
+	change_plant(plant)
+	change_color()
+
+	update_icon()
+
+	set_light(brightness_on, brightness_on/2)
+
+/obj/structure/cyberplant/update_icon()
+	..()
+	overlays.Cut()
+	if (!plant)
+		return
+
 	overlays += plant
-	set_light(3)
+
+/obj/structure/cyberplant/proc/change_plant(var/state)
+	plant = prepare_icon(state)
+
+/obj/structure/cyberplant/proc/change_color(var/color)
+	if (!plant)
+		return
+
+	if(!color)
+		color = pick(possible_colors)
+
+	light_color = color
```

Comment:
Editing `light_color` directly is bad. You should use `set_light(l_color=color)` instead.

Reply:
Done.